### PR TITLE
exposed new search feature in sdk

### PR DESCRIPTION
**Pull request by royWeisselberg**
Added the ability to search for multiple keys using a single term

### CODE REVIEW

File: tests/storage/storage.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            backendOnly: false,
+          },
+          {
+            key: 'aaac',
+            value: 3,
+            backendOnly: false,
+          },
+        ],
+      ],
+    ])('Should return records matching the searched term', async records => {
```
Comment (gregra81) [inline tests/storage/storage.test.ts:36]: what's the value of this test?

File: lib/storage/base-storage.ts [hunk 2]
```diff
@@ -1,100 +1,107 @@
...
+  public counterUrl() {
+    const storageUrl = this.getStorageUrlV2();
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    const fullPath = `${storageUrl}/operations`;
+    return fullPath;
+  }
+
+  public searchUrl(key: string, options: SearchOptions) {
+    const storageUrl = this.getStorageUrlV2();
+    const cursor = options.cursor ? `&cursor=${options.cursor}` : '';
```
Comment (DorShakedMonday) [inline lib/storage/base-storage.ts:103]: Do we control `cursor`? is it validated and sanitized in `storage-api` side?

File: lib/storage/storage.ts [hunk 3]
```diff
@@ -27,22 +30,37 @@ export class Storage extends BaseStorage implements IStorageInstance {
...
+
+  async search<T extends JsonValue>(key: string, options: SearchOptions = {}): Promise<SearchResponse<T>> {
+    const url = this.searchUrl(key, options);
+    const params = { method: 'GET' };
+    const result = await this.storageFetchV2<SearchServerResponse<T>>(url, params);
+    if (!isDefined(result)) {
+      return { success: false, records: null };
+    }
+
+    const response: SearchResponse<T> = {success: true, records: result.records};
```
Comment (DorShakedMonday) [inline lib/storage/storage.ts:49]: Your prettier/linter is working partially.
This creates a strange un-even formatting 😅

File: tests/storage/storage.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  describe('Search', () => {
+    it.each([
+      [{}, `https://apps-storage.monday.com/api/v2/items?term=${term}`],
+      [{ cursor: cursorID }, `https://apps-storage.monday.com/api/v2/items?term=${term}&cursor=${cursorID}`],
+    ])(
+      'Should show correct URL when called with a term and return array for result',
+      async (cursor: object, expectedUrl: string) => {
+        // Arrange
+        const storage = new Storage(FAKE_TOKEN);
+
```
Comment (gregra81) [inline tests/storage/storage.test.ts:38]: add comment that this is mocking the storage api service

File: tests/storage/storage.test.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        // Assert
+        expect(results.success).toEqual(true);
+        expect(results.records).toBeInstanceOf(Array);
+        expect(fetch).toHaveBeenCalledTimes(1);
+        expect(fetch).toHaveBeenCalledWith(expectedUrl, expectedOptions);
+      },
+    );
+
+    it.each([[StatusCodes.NO_CONTENT], [StatusCodes.NOT_FOUND]])(
```
Comment (gregra81) [inline tests/storage/storage.test.ts:55]: Not required

**Review by gregra81 — APPROVED** [review]
Looks great

**Pull request merged**